machine/models/baseModel.py

```diff
@@ -0,0 +1,49 @@
+import torch.nn as nn
+import torch.nn.functional as F
+
+class Model(nn.Module):
+    """ 
+    Abstract base class for models.
+
+    Args:
+        encoder_module (baseRNN) :  module that encodes inputs
+        decoder_module (baseRNN, optional):   module to decode encoded inputs
```

Comment:
Shouldn't the basemodel be agnostic about the underlying model for encoding and decoding?
The encoder could very well be a CNN instead of `baseRNN`.

Reply:
Good catch. machine is in principle a sequential library but it doesn't hurt including this case for future reference. As long as we specify what the input-output behaviour is in could be any module, only CNN has different io behaviour than a sequential module, right? This may complicate things.

Reply:
I think we can make it work, but definitely might have to put more thinking if we want this. SO the obvious question is do we want this?

Reply:
I think it would be nice to have it but at the same time I'm not keen on having to write 15 times as many docstrings to make it clear what's going on. Perhaps we can put it on the TODO list for the future? Maybe a student could implement it in a project?

Reply:
Agreed

Reply:
I created an issue with label feature request for this, so that we can do it in the future.